tests/test-customize-partials-controller.php

```diff
@@ -0,0 +1,313 @@
+		$this->assertEquals( array( 'view', 'embed', 'edit' ), $data['endpoints'][0]['args']['context']['enum'] );
+
+		// Test single.
+		$request = new WP_REST_Request( 'OPTIONS', '/customize/v1/partials/test' );
+		$response = $this->server->dispatch( $request );
+		$data = $response->get_data();
+		$this->assertEquals( 'view', $data['endpoints'][0]['args']['context']['default'] );
+		$this->assertEquals( array( 'view', 'embed', 'edit' ), $data['endpoints'][0]['args']['context']['enum'] );
+	}
+
+	/**
+	 * Test get_item_schema.
+	 *
+	 * @covers WP_REST_Customize_Partials_Controller::get_item_schema()
+	 */
+	public function test_get_item_schema() {
+		$changeset_controller = new WP_REST_Customize_Partials_Controller();
+		$schema = $changeset_controller->get_item_schema();
+		$properties = $schema['properties'];
+
+		$this->assertSame( 6, count( $properties ) );
+
+		$this->assertArrayHasKey( 'fallback_refresh', $properties );
+		$this->assertSame( 'boolean', $properties['fallback_refresh']['type'] );
+
+		$this->assertArrayHasKey( 'container_inclusive', $properties );
+		$this->assertSame( 'boolean', $properties['container_inclusive']['type'] );
+
+		$this->assertArrayHasKey( 'selector', $properties );
+		$this->assertSame( 'string', $properties['selector']['type'] );
+
+		$this->assertArrayHasKey( 'id', $properties );
+		$this->assertSame( 'string', $properties['id']['type'] );
+
+		$this->assertArrayHasKey( 'settings', $properties );
+		$this->assertSame( 'array', $properties['settings']['type'] );
+
+		$this->assertArrayHasKey( 'type', $properties );
+		$this->assertSame( 'string', $properties['type']['type'] );
+	}
+
+	/**
+	 * Test get_item.
+	 *
+	 * @covers WP_REST_Customize_Partials_Controller::get_item()
+	 */
+	public function test_get_item() {
+		wp_set_current_user( self::$admin_id );
+
+		$request = new WP_REST_Request( 'GET', sprintf( '/customize/v1/partials/%s', self::TEST_SETTING_ID ) );
+
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 200, $response->get_status() );
+		$data = $response->get_data();
+
+		$this->assertSame( self::TEST_SETTING_ID, $data['id'] );
+	}
+
+	/**
+	 * Filter partial args.
+	 *
+	 * @param bool   $args If add dynamic partial.
+	 * @param string $partial_id Partial ID.
+	 * @return bool If add dynamic partial.
+	 */
+	public function filter_customize_dynamic_partial_args( $args, $partial_id ) {
+		if ( self::TEST_SETTING_ID === $partial_id ) {
+			return true;
+		}
+		return $args;
+	}
+
+	/**
+	 * Test getting dynamic partial with 'customize_dynamic_partial_args'.
+	 */
+	public function test_get_item_with_dynamic_partial_filter() {
+		add_filter( 'customize_dynamic_partial_args', array( $this, 'filter_customize_dynamic_partial_args' ), 10, 2 );
+
+		wp_set_current_user( self::$admin_id );
+
+		$request = new WP_REST_Request( 'GET', sprintf( '/customize/v1/partials/%s', self::TEST_SETTING_ID ) );
```

Comment:
Actually `test_get_item_invalid_id` is testing a missing partial. Guess the name `test_get_item_missing` would be a more accurate name for the test.